src/geshi.php

```diff
@@ -611,7 +611,7 @@ public function error() {
      */
     public function get_language_name() {
         if (GESHI_ERROR_NO_SUCH_LANG == $this->error) {
-            return $this->language_data['LANG_NAME'] . ' (Unknown Language)';
```

Comment:
Better to have this set even in case of errors so there's no need for every location to do some special check for such situations.

Reply:
Yes. Having one line access `$this->language`, while everything else accesses `$this->language_data['LANGUAGE_NAME']` seems rather inconsistent. And as most of the parser assumes there's some (somewhat) consistent structure in `$this->langauge_data` anyway we just might opt to have `set_language` do that setup instead if some error happens.